### DISCUSSION TITLE
New (almost) empty nuxt 3.4 project, entry.js is almost 100kb?

### DISCUSSION THREAD

**Discussion by jorismak** [nuxt/nuxt#20230] [Questions] [answered]
Hi there,

This is just a question to see if this is normal, or if there is something weird.

I understand that nuxt has to include the Vue runtime and maybe some smaller libraries, but still I find 100kb (well, 97.6kb) quite a lot 'to get started'.

I just used `nuxi init` to create a new Nuxt 3 project. I then replaced the welcome component in app.vue, with a simple template like this:
```
<template>
    <div>
        <h1>An empty page</h1>
    </div>
</template>
```

So, this is all there is. No plugins, no other libraries, no css, etc...

Doing `pnpm build` generates the heavy entry js file in the client bundle:
```
.nuxt/dist/client/_nuxt/entry.b2d6f8dd.js            97.24 kB │ gzip: 37.08 kB
```

Looking at this by serving it and/or previewing it, the entry.js indeed gets sent to the browser (just to check if the file is actually being used).

Doing a `nuxt analyze`, shows me for the client bundle that the biggest is @vue/runtime-core (which makes sense), but it's < 52kb.
I also see a @vue/runtime-dom of 5.6kb and a @vue/reactivity that is 13.19kb.

That makes +/- 70kb for Vue3. Is Vue3 runtime-without-template-compiler really that big these days?

Still, that adds 9.20kb for unhead, with another 3.65 for @unhead/dom and another 1.4kb for @unhead/shared, and another 4kb for ufo.

Everything that seems node_modules/nuxt related is only 10kb combined. So I have a feeling there are big libraries being included that maybe don't get tree-shaked completely?

Why is unhead so big for instance (14.25kb combined, smaller than the whole Nuxt runtime?), and why is there something for h3 included in the client-bundle (node_modukes/h3/dist/index.mjs)?

Things that seem logical, but I didn't realize from the start are things like the error component (and the 404 and 500 page), that are 1.404 + 2kb + 2.4kb = give or take 5.8kb. But these are included in the main 'entry' bundle. Can't they be loaded when needed? Specially the 500 and 404 implementation (I can understand a base 'error component' is needed internally for knowing how to deal with errors in loading an async error page :)).

'nuxt-link' itself also is 3.5kb, but maybe there is some router code included in that?

Again, I'm not here to complain. I'm new to the SRR world, and I'm just wondering what is expected to be normal. From reading online and reading some old issues from 2022 I seem to see smaller sizes by users, buy maybe those are gzipped already?

**Comment by kaikaibenkai** [comment]
&lt;deleted&gt;

**Comment by TheAlexLichter** [comment, marked as answer]
You can find the bundle size tests in https://github.com/nuxt/nuxt/blob/main/test/bundle.test.ts

So yes, 97.2 kB (uncompressed) is the current size of an application, shrinking down to roughly one third (~37kB) of it when applying state of the art compression, which should happen in production.

Be aware that Vue.js itself (full build using all features) is alone a 34kB gzipped (see https://bundlephobia.com/package/vue@3.3.4)

**Comment by jorismak** [comment]
Good to be aware of, thanks. It seemed like a lot for an empty project, but 1) it's not me doing anything wrong apparently, and 2) considering vue runtime size it's not bad at all.

I had a feeling Vue was smaller when just picking the runtime? Guess I have that wrong in my memory (I see on other bundle-size-comparisons now that 20kb for Vue2 gzipped was a thing, but 30 to 34kb gzip'd for vue3.3 is quite normal).

So, to answer my own question:

> That makes +/- 70kb for Vue3. Is Vue3 runtime-without-template-compiler really that big these days?

Yes, yes it is.
